src/Commands/Domain/SetPrimaryCommand.php

```diff
@@ -0,0 +1,124 @@
+
+/**
+ * Class SetPrimaryCommand
+ * @package Pantheon\Terminus\Commands\Domain
+ */
+class SetPrimaryCommand extends TerminusCommand implements SiteAwareInterface
+{
+    use SiteAwareTrait;
+    use WorkflowProcessingTrait;
+
+    /**
+     * Sets the primary domain for a site and environment, causing all traffic to redirect to the primary domain.
+     *
+     * @authorize
+     *
+     * @command domain:primary:set
+     *
+     * @param string $site_env Site & environment in the format `site-name.env`
+     * @param string $domain A domain that has been associated to your site. Optional when running interactively.
+     *
+     * @usage domain:primary:set <site_env>
+     */
+    public function set($site_env, $domain)
+    {
+        /**
+         * @var $site Site
+         * @var $env Environment
+         */
+        list(, $env) = $this->getSiteEnv($site_env);
+
+        // The primary domain is set via a workflow so as to use workflow logging to track changes.
+        $this->log()->notice('Setting primary domain to {domain}...', ['domain' => $domain]);
+        $this->runWorkflowForEnvironmentAndDomain($env, $domain);
+    }
+
+    /**
+     * Unsets the primary domain for a site and environment.
+     *
+     * @authorize
+     *
+     * @command domain:primary:unset
+     *
+     * @param string $site_env Site & environment in the format `site-name.env`
+     */
+    public function reset($site_env)
+    {
+        /**
+         * @var $site Site
+         * @var $env Environment
+         */
+        list(, $env) = $this->getSiteEnv($site_env);
+
+        $this->log()->notice('Unsetting primary domain...');
+        $this->runWorkflowForEnvironmentAndDomain($env, null);
+    }
+
+    protected function runWorkflowForEnvironmentAndDomain(Environment $env, $domain)
+    {
+        /** @var Workflow */
+        $workflow = $env->getWorkflows()->create(
+            'set_primary_domain',
+            [
+                'environment' => $env->id,
+                'params'      => ['primary_domain' => $domain]
+            ]
+        );
+        $this->processWorkflow($workflow);
+    }
+
+    /**
+     * Prompt the user for the domain, if it was not specified.
+     *
+     * n.b. This hook is not called in --no-interaction mode.
+     *
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @param AnnotationData $annotationData
+     *
+     * @hook interact domain:primary:set
+     */
+    public function setInteract(InputInterface $input, OutputInterface $output, AnnotationData $annotationData)
+    {
+        $domain = $input->getArgument('domain');
+        if (empty($domain)) {
+            /**
+             * @var $site Site
+             * @var $env Environment
+             */
+            list($site, $env) = $this->getSiteEnv($input->getArgument('site_env'));
+            $domains = $env->getDomains()->ids();
+            // Put pantheonsite.io domains at the bottom.
```

Comment:
I think it would be better to prohibit setting `pantheonsite.io` as a primary domain, and just filter these out of the list.

Reply:
That's kind of a business rule implemented client-side. I agree it seems odd that you would set one of these domains as primary, but it's currently supported by the platform and if you can't do that, Terminus shouldn't be the only place that says no you can't do that. It's trivial to filter; can we remove it as a follow-up and talk to the rest of Core about whether to deny this?

Reply:
The `getDomains` API returns all of the available domains, including the platform domains. We can't modify this API to not return platform domains, or we'd break other clients.

We could add an API `isPlatformDomain` or something like that. The two places that would need to use that API is this PR, and the Dashboard user interface for setting the primary domain, which hasn't been written yet.

The existing code already has explicit knowledge of what the platform domain is, in order to sort the platform domain to the end of the list. This logic, if maintained, would also need to be duplicated in the Dashboard, and could also benefit from an API to identify the platform domain (or even to sort the domain list).

I tend to think that these very specific APIs are not necessary, and that it would be okay to put this business logic in the two clients that need it. I think it's fine to raise the issue to `@core` and collect feedback from a wider audience. I think that filtering is better than changing the sort order, but I care 3/10.

Reply:
Oh, I didn't mean to change the list APIs, just that if we filter then I think the workflow ought to fail if asked to make a pantheon.io domain primary.

Reply:
I decided I was thinking with my developer hat, and not my UX hat. From the perspective of the "it is not easy to inadvertently set the wrong primary domain" perspective, filtering here but leaving it supported on the platform is exactly the right solution, as I discuss in most recent commit message.